tests/cypress/tests-e2e/geolocation.cy.js

```diff
@@ -49,54 +65,104 @@ describe('Geolocation cypress', () => {
+                })
+                // check that the map has been centered on the geolocation and zoom is updated
+                cy.readStoreValue('state.position.center').then((center) => {
+                    expect(center).to.be.an('Array')
+                    expect(center.length).to.eq(2)
+                    expect(center[0]).to.approximately(geoX, 0.1)
+                    expect(center[1]).to.approximately(geoY, 0.1)
+                })
+                cy.readStoreValue('state.position.zoom').then((zoom) => {
+                    expect(zoom).to.eq(SWISS_ZOOM_LEVEL_1_25000_MAP)
                 })
             })
-        }
-    )
+            it('access from outside Switzerland shows an error message', () => {
+                // null island
+                cy.goToMapView({}, true, { latitude: 0, longitude: 0 })
+                getGeolocationButtonAndClickIt()
+                testErrorMessage('geoloc_out_of_bounds')
 
-    context(
-        'Test geolocation when geolocation is unauthorized',
-        {
-            env: {
-                browserPermissions: {
-                    geolocation: 'block',
-                },
-            },
-        },
-        () => {
-            it('shows an alert telling the user geolocation is unauthorized when the geolocation button is clicked', () => {
-                cy.goToMapView()
+                // Java island
+                cy.goToMapView({}, true, { latitude: -7.71, longitude: 110.37 })
                 getGeolocationButtonAndClickIt()
-                cy.on('window:alert', (txt) => {
-                    expect(txt).to.contains(
-                        'The acquisition of the position failed because your browser settings does not allow it. Allow your browser /this website to use your location. Deactivate the "private" mode of your browser'
-                    )
-                })
+                testErrorMessage('geoloc_out_of_bounds')
             })
         }
     )
+
+    context('Test geolocation when geolocation is failed to be retrieved', () => {
+        it('shows an error telling the user geolocation is denied', () => {
+            cy.goToMapView({}, true, { errorCode: GeolocationPositionError.PERMISSION_DENIED })
+            getGeolocationButtonAndClickIt()
+            testErrorMessage('geoloc_time_out')
+        })
+
+        it('shows an alert telling the user geolocation is not able to be retrieved due to time out', () => {
+            cy.goToMapView({}, true, { errorCode: GeolocationPositionError.TIMEOUT })
+            getGeolocationButtonAndClickIt()
+            testErrorMessage('geoloc_time_out')
+        })
+        it('shows an alert telling the user geolocation is not available for other reason', () => {
+            cy.goToMapView({}, true, { errorCode: GeolocationPositionError.POSITION_UNAVAILABLE })
+            getGeolocationButtonAndClickIt()
+            testErrorMessage('geoloc_unknown')
+        })
+    })
```

Comment:
you can wrap up all these 3 its into a single one, that would lower our it count (that's our hard limit with CypressCloud (we have 100k tests a month, a test is a `it()` block)

Reply:
unfortunately, I can only reduce the number of `it` from 3 to 2. I couldn't find a way to put the one testing `POSITION_UNAVAILABLE` in the same `it` block as the other 2. I have tried several tricks to do it (e.g. `cy.reload`, `restore stub`) but it still failed.

Reply:
looks like it breaks a bit your testing to split them off...
Then never mind, keep them separated

Reply:
yep, when I put into 2 blocks, it pass locally but not in CI. I change it back to 3 blocks now.